src/App.tsx

```diff
@@ -1,26 +1,266 @@
+         2   , null, null, null, 9   , null, 5   , null, 7   ,
+         7   , null, 3   , 2   , null, null, 6   , null, 1   ] as Array<SquareValue | null>;
   return (
     <div className="App">
       <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <Board values={vals} />
       </header>
     </div>
   );
+};
+
+type SquareAddress = number;
+
+interface BoardState {
+    highlights: SquareAddress[];
+    contradicts: SquareAddress[];
+    permanentValues: SquareAddress[];
+    values: Array<SquareValue | null>;
+}
+
+interface BoardProps {
+    values: Array<SquareValue | null>;
+}
+
+class Board extends React.Component<BoardProps, BoardState> {
+    public state: BoardState = {
+        highlights: [37],
+        contradicts: [38],
+        permanentValues: [],
+        values: this.props.values,
+        /**
+          * //TODO: CREATE marking state and transfer marking logic to board state.
+          */
+    };
+
+    /**
+      * // TODO: other highlight functionality
+      */
+    public toggleSelectSquare(i: SquareAddress) {
+        this.setState( {
+            ...this.state,
+            highlights: [i],
+            },
+        );
+    }
+
+    public getPermanentValues(): SquareAddress[] {
+        const list = [];
+        for (let i = 0; i < 81; i++) {
+            if (this.props.values[i]) {
+                list.push(i as SquareAddress);
+            }
+        }
+        return list;
+    }
+
+    public normalMarkSelectedSquares(i: SquareValue) {
+        for (let j = 0; j < this.state.highlights.length; j++) {
+            const addy: SquareAddress = this.state.highlights[j];
+            if (!this.state.permanentValues.includes(addy)) {
+                const newValues = JSON.parse(JSON.stringify(this.state.values));
+                newValues[addy] = i;
+                this.setState({
+                    values: newValues,
+                });
+            }
+        }
+    }
+
+    public deleteSelectedSquares(): void {
+        const newValues = JSON.parse(JSON.stringify(this.state.values));
+        for (const address of this.state.highlights) {
+            if (!this.state.permanentValues.includes(address)) {
+                newValues[address] = null;
+                /**
+                  * //TODO: if marking exist on first delete click
+                  * delete only value and show markings on second
+                  * click delete markings
+                  */
+            }
+        }
+        this.setState({
+            values: newValues,
+        });
+    }
+
+/*     convertUndefinedValues(): (SquareValue | null)[] {
```

Comment:
Commented code should not be committed. Delete it and refer to the commit history to retrieve it.